src/economy.cpp

```diff
@@ -1757,7 +1757,7 @@ static void LoadUnloadVehicle(Vehicle *front)
 				break;
 
 			case VEH_ROAD:
-				t = front->vcache.cached_max_speed / 2;
+				t = front->vcache.cached_max_speed;
 				break;
```

Comment:
This switch case can be merged with the one immediately above it now that both share the same code, creating one block as follows:
```c++
case VEH_TRAIN:
case VEH_ROAD:
case VEH_SHIP:
	t = front->vcache.cached_max_speed;
	break;
```

Reply:
I would also appreciate a comment (for ships as well) which explains why the conversion of half-units to units is deliberately not done, rather than possibly just forgotten about.

Reply:
I find it hard to explain what is hapening here and I agree that the code does not really reflect what is hapening when the max_speed is alreaddy double in size before this function is called. But i have inserted a comment and made the optimizations as suggested.